packages/paste-core/components/form-pill-group/src/FormPillGroup.tsx

```diff
@@ -39,26 +44,65 @@ export interface FormPillGroupProps
    * @memberof FormPillGroupProps
    */
   display?: "flex" | "inline-flex";
+  size?: FormPillGroupSizeVariant;
+  setPillFontSize: Dispatch<SetStateAction<string|undefined>>;
+
+}
+
+/** The response from the calculateSizeStyles function. */
+type CalculateSizeStylesResponse = Pick<BoxProps, "columnGap" | "rowGap" | "fontSize">;
+
+/**
+ * Calculates the styles for the FormPillGroup based on the size prop.
+ *
+ * Exported only to be used in tests.
+ * @private
+ */
+export function calculateSizeStyles(size: FormPillGroupSizeVariant): CalculateSizeStylesResponse {
+  if (size === "L") {
+    return {
+      columnGap : "space30",
+      rowGap : "space30",
+      fontSize : "fontSize30",
+    }
+  }
+  return {
+    columnGap : "space20",
+    rowGap : "space20",
+    fontSize : undefined,
+  }
 }
 
 const FormPillGroupStyles = React.forwardRef<HTMLUListElement, FormPillGroupProps>(
-  ({ element = "FORM_PILL_GROUP", display = "flex", ...props }, ref) => (
-    <Box
-      {...safelySpreadBoxProps(props)}
-      element={element}
-      ref={ref}
-      role="listbox"
-      lineHeight="lineHeight30"
-      margin="space0"
-      padding="space0"
-      display={display}
-      flexWrap="wrap"
-      rowGap="space20"
-      columnGap="space20"
-    >
-      {props.children}
-    </Box>
-  ),
+  ({ element = "FORM_PILL_GROUP", display = "flex", size = "default", ...props }, ref) => {
+    const {
+      columnGap,
+      rowGap,
+      fontSize,
+    } = calculateSizeStyles(size)
+    useEffect(() => {
+      props.setPillFontSize(fontSize)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
```

Comment:
I don't love this effect hook, but it seemed like the most Paste consumer friendly way to calculate the fontSize based on the size property and automatically apply it to ever FormPill.

Reply:
Effect hook now gone thanks to feedback.